src/primitive_crypto.cc

```diff
@@ -24,6 +24,45 @@
 
 namespace toit {
 
+enum class AesTypes {
+  Ecb,
+  Cbc
+};
+
+static Object* aes_crypt(Process* process, AesEcbContext* context, Blob* input, int from, int to, bool encrypt, AesTypes type) {
+  if (from < 0 || to > input->length() || from > to || ((to - from) % 16) != 0) INVALID_ARGUMENT;
+
+  Error* error = null;
+  ByteArray* result = process->allocate_byte_array(to - from, &error);
+  if (result == null) return error;
+
+  ByteArray::Bytes output_bytes(result);
+
+  switch (type)
+  {
+  case AesTypes::Ecb:
+    mbedtls_aes_crypt_ecb(
+      &context->context_,
+      encrypt ? MBEDTLS_AES_ENCRYPT : MBEDTLS_AES_DECRYPT,
+      input->address() + from,
+      output_bytes.address());
+    break;
+  case AesTypes::Cbc:
+    mbedtls_aes_crypt_cbc(
+      &context->context_,
+      encrypt ? MBEDTLS_AES_ENCRYPT : MBEDTLS_AES_DECRYPT,
+      to - from,
+      static_cast<AesCbcContext*>(context)->iv_,
+      input->address() + from,
+      output_bytes.address());
+      break;
```

Comment:
Less indentation.

Reply:
Made the indentation 2 spaces like the rest of the file.